src/jdk.incubator.jextract/share/classes/jdk/internal/clang/libclang/RuntimeHelper.java

```diff
@@ -78,17 +79,28 @@ private RuntimeHelper() {}
 
     static final MemorySegment lookupGlobalVariable(LibraryLookup[] LIBRARIES, String name, MemoryLayout layout) {
         return lookup(LIBRARIES, name).map(s ->
-            nonCloseableNonTransferableSegment(s.address().asSegmentRestricted(layout.byteSize())
-                    .share())).orElse(null);
+            nonCloseableNonTransferableSegment(s.address().asSegmentRestricted(layout.byteSize()))).orElse(null);
     }
 
+    final static SegmentAllocator DEFAULT_ALLOCATOR = MemorySegment::allocateNative;
+
     static final MethodHandle downcallHandle(LibraryLookup[] LIBRARIES, String name, String desc, FunctionDescriptor fdesc, boolean variadic) {
         return lookup(LIBRARIES, name).map(
                 addr -> {
                     MethodType mt = MethodType.fromMethodDescriptorString(desc, LOADER);
                     return variadic ?
                         VarargsInvoker.make(addr, mt, fdesc) :
-                        LINKER.downcallHandle(addr, mt, fdesc);
+                        LINKER.downcallHandle(addr, DEFAULT_ALLOCATOR, mt, fdesc);
+                }).orElse(null);
+    }
+
+    static final MethodHandle downcallHandle(LibraryLookup[] LIBRARIES, String name, String desc, FunctionDescriptor fdesc, boolean variadic, NativeScope scope) {
```

Comment:
What do you think of this? This method still seems unused in the latest version

Reply:
This is generated code, I don't think it's worth making changes on code that will be overwritten when we regenerate this. There are loads of methods that are unused in the libclang port (because jextract doesn't use all libclang functions).

Reply:
Ah ok, I thought this was added by hand. Sorry